src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -95,6 +95,22 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
                 _logger.LogInformation("Received {textEditsLength} results from C#.", textEdits.Length);
             }
 
+            // Sometimes teh C# document is out of sync with our document, so Roslyn can return edits to us that will throw when we try
+            // to normalize them. Instead of having this flow up and log a NFW, we just capture it here. Since this only happens when typing
+            // very quickly, it is a safe assumption that we'll get another chance to do on type formatting, since we know the user is typing.
+            // The proper fix for this is https://github.com/dotnet/razor-tooling/issues/6650 at which point this can be removed
+            foreach (var edit in textEdits)
+            {
+                var startLine = edit.Range.Start.Line;
+                var endLine = edit.Range.End.Line;
+                var count = csharpText.Lines.Count;
+                if (startLine >= count || endLine >= count)
```

Comment:
The logic here comes direct from https://github.com/dotnet/razor-tooling/blob/main/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/RangeExtensions.cs#L131-L139 which is what is throwing the exceptions. I decided to leave that in place though, and just fix it here, because, as per the comment, I think on type formatting is pretty low stakes in terms of consequences of skipping one, but the exception could still help us identify other issues in future.

Reply:
Is it ever the case that startLine > endLine? i.e. can the check be simplified to `if (endLine >= count)` instead?

Reply:
Logically there could be, and mainly I wanted to mirror the existing logic that would result in an NFW, but I certainly couldn't repro it.